packages/abi-gen/test-cli/fixtures/contracts/AbiGenDummy.sol

```diff
@@ -138,4 +138,90 @@ contract AbiGenDummy
     uint someState;
     function nonPureMethod() public returns(uint) { return someState += 1; }
     function nonPureMethodThatReturnsNothing() public { someState += 1; }
+
```

Comment:
We've been keeping all the test cases in one contract to speed up test run times. I'm thinking about how we could keep it somewhat organised, since we're working towards having full coverage of Solidity language features. We don't have this convention yet, but what about prefixing a section of test cases with the methods it's meant to test?

e.g. 
```
// begin tests for `decodeTransactionData`, `decodeReturnData`
...
// end tests for `decodeTransactionData`, `decodeReturnData`
```

Reply:
Why not just use inheritance?

Reply:
@xianny I've added comments for now for consistency, although I agree w Dorothy - it could be good to split this into multiple contracts alongside the refactor of `abi-gen`.

Reply:
Ya, let's keep that in mind. I'm a little concerned that inheritance might complicate testing, especially when we add the EVM simulator. As in we would have to test whatever case we want PLUS inheritance.